### PR TITLE
Remove warnings due to some captured variables

### PR DESCRIPTION
**Pull request by atrantan**
This patch tries to remove some warnings related to unused captured variables detected by Clang 5.0. Because most of those variables are copied via lambda capture for lifetime purpose, HPX_UNUSED() has been used. Also warnings are generated when local const variables are captured by value;  some of those variables are then declared non-const in place of const.

### CODE REVIEW

File: tests/unit/parallel/algorithms/reduce_.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ void test_reduce1(ExPolicy policy, IteratorTag)
     std::vector<std::size_t> c(10007);
     std::iota(boost::begin(c), boost::end(c), std::rand());
 
-    std::size_t const val(42);
+    std::size_t val(42);
```
Comment (hkaiser) [inline tests/unit/parallel/algorithms/reduce_.cpp]: What's the rationale of this change?

File: tests/unit/parallel/algorithms/reduce_.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ void test_reduce1_async(ExPolicy p, IteratorTag)
...
```
Comment (hkaiser) [inline tests/unit/parallel/algorithms/reduce_.cpp]: What's the rationale of this change?

File: tests/unit/parallel/algorithms/transform_reduce.cpp [hunk 3]
```diff
@@ -79,7 +79,7 @@ void test_transform_reduce_async(ExPolicy p, IteratorTag)
     std::vector<std::size_t> c(10007);
     std::iota(boost::begin(c), boost::end(c), std::rand());
 
-    std::size_t const val(42);
+    std::size_t val(42);
```
Comment (hkaiser) [inline tests/unit/parallel/algorithms/transform_reduce.cpp]: What's the rationale of this change?

File: tests/unit/parallel/algorithms/reduce_.cpp (hunk 1 above)
Comment (atrantan) [inline tests/unit/parallel/algorithms/reduce_.cpp]: Got this strange error from clang 5.0 although the variable is used inside the lambda : `warning: lambda capture 'val' is not required to be captured for this use [-Wunused-lambda-capture]`

File: tests/unit/parallel/algorithms/reduce_.cpp (hunk 2 above)
Comment (atrantan) [inline tests/unit/parallel/algorithms/reduce_.cpp]: Same issue

File: tests/unit/parallel/algorithms/transform_reduce.cpp (hunk 3 above)
Comment (atrantan) [inline tests/unit/parallel/algorithms/transform_reduce.cpp]: Same issue
